microcosm_flask/conventions/upload.py

```diff
@@ -0,0 +1,136 @@
+"""
+Conventions for file upload.
+
+"""
+from contextlib import contextmanager
+from os.path import join
+from shutil import rmtree
+from tempfile import mkdtemp
+
+from flask import request
+from marshmallow import Schema
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.conventions.encoding import (
+    dump_response_data,
+    load_query_string_data,
+    merge_data,
+)
+from microcosm_flask.conventions.registry import qs, response
+from microcosm_flask.operations import Operation
+from werkzeug.exceptions import BadRequest
+from werkzeug.utils import secure_filename
+
+
+try:
+    # Python 2
+    from contextlib import nested
+except ImportError:
+    # Python3
+    from contextlib import ExitStack
+
+    @contextmanager
+    def nested(*contexts):
+        """
+        Reimplementation of nested in python 3.
+        """
+        with ExitStack() as stack:
+            results = [
+                stack.enter_context(context)
+                for context in contexts
+            ]
+            yield results
+
+
+@contextmanager
+def temporary_upload(name, fileobj):
+    """
+    Upload a file to a temporary location.
+
+    Flask will not load sufficiently large (>500M) files into memory, so it
```

Comment:
only reference for limit I could find was [this](http://stackoverflow.com/a/15782516/1152258) which stated 500KB rather than 500M